src/app/month-calendar/month-calendar.service.ts

```diff
@@ -38,9 +38,9 @@ export class MonthCalendarService {
 
   generateYear(config: IMonthCalendarConfig, year: Moment, selected: Moment[] = null): IMonth[][] {
     const index = year.clone().startOf('year');
-
-    return this.utilsService.createArray(3).map(() => {
-      return this.utilsService.createArray(4).map(() => {
+    const inRow = config.monthInRow && 12 % config.monthInRow === 0 ? config.monthInRow : 4;
```

Comment:
please extract it to a private function.

Reply:
Ok